### ISSUE TITLE
static can be executed twice (or more), if containing function is called again while it's initializing

### ISSUE THREAD

**Issue by gnysek** [YoYoGames/GameMaker-Bugs#7352]
### Description
I've found an edge case in which static property of function will be executed twice.

It happens when while initializing static value, function which does it will be called again.

In attached example, we have two functions - `StaticTest` and `__System`.

By calling `StaticTest()` it's static variable `_testData` is intializing, but inside that initialization, we're calling `__System`, which also calls `StaticTest()`, and that results in calling it another time.

From my point, it looks that at this stage static `_testData` looks like unitialized yet, so it's called again.

Example code (attached to this ticket):
```gml
function StaticTest() {
	static _testData = (function(){
		show_debug_message("static function executed");
		show_debug_message(debug_get_callstack());
		
		__System();
		
		return true;
	})(); // <-- notice brackets here
}

function __System() {
	static _system = undefined;
    if (_system != undefined) return _system;
	
	_system = {};
	
	StaticTest();
	
	return _system;
}

show_debug_message("=================");
StaticTest();
```
and result log:
```
=================
static function executed
[ "gml_Script__testData@anon@45@StaticTest@Script1:4","gml_Script_StaticTest:2","gml_GlobalScript_Script1:25",0 ]
static function executed
[ "gml_Script__testData@anon@45@StaticTest@Script1:4","gml_Script_StaticTest:2","gml_Script___System:19","gml_Script__testData@anon@45@StaticTest@Script1:6","gml_Script_StaticTest:2","gml_GlobalScript_Script1:25",0 ]
```

### Steps To Reproduce
1. Run example code

### Which version of GameMaker are you reporting this issue for?
IDE v2024.800.0.609 Runtime v2024.800.0.633

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

[16de9f23-fd37-4eed-8a23-81d0ca6c9d5f](https://api.gamemaker.io/api/github/downloads/16de9f23-fd37-4eed-8a23-81d0ca6c9d5f)

**Comment by gnysek** [comment]
Update: seems to not work in GMS2 VM and GMS2 YYC, but works in GMRT.

**Comment by rwkay** [comment]
Fixed in 2024.10

**Comment by scott-dunbar** [comment]
Verified fixed in  IDE v2024.1100.0.639  Runtime v2024.1100.0.662
